x/axelarnet/keeper/msg_server.go

```diff
@@ -416,27 +416,25 @@ func (s msgServer) RetryIBCTransfer(c context.Context, req *types.RetryIBCTransf
 func (s msgServer) ExecuteMessage(c context.Context, req *types.ExecuteMessageRequest) (*types.ExecuteMessageResponse, error) {
 	ctx := sdk.UnwrapSDKContext(c)
 
-	chain, ok := s.nexus.GetChain(ctx, req.ID.Chain)
+	msg, ok := s.nexus.GetMessage(ctx, req.ID)
 	if !ok {
-		return nil, fmt.Errorf("invalid chain %s", req.ID.Chain)
+		return nil, fmt.Errorf("message %s not found", req.ID)
 	}
 
-	if !s.nexus.IsChainActivated(ctx, chain) {
-		return nil, fmt.Errorf("chain %s is not activated", chain.Name)
+	if !msg.Sender.Chain.IsFrom(evmtypes.ModuleName) {
+		return nil, fmt.Errorf("message is not from an EVM chain")
```

Comment:
I know this is from before, but do we want to restrict the source to evm chains? What if we start supporting another type of chain besides evm and cosmos?

Reply:
it is currently for abi encoding translation, so we put a explicit check. 
If we have more chain types, we will either remove the translation to make it generic, or update the validation here.